File: ExpenseTracker.Domain/Entities/Expense.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExpenseTracker.Domain.Entities
{
    public class Expense
    {
        public int Id { get; set; }
        public string Description { get; set; }
        public DateOnly TransactionDate { get; set; }
        public decimal Value { get; set; }

        public int AccountId { get; set; }
        public Account Account { get; set; }

        public int CategoryId { get; set; }
        public Category Category { get; set; }

        public int ExpenseTypeId { get; set; }
        public ExpenseType ExpenseType { get; set; }

        public string? BillImgUrl { get; set; }
        
    }


}

File: ExpenseTrackerView/src/app/interfaces/expense.ts

export interface ExpenseSend {
    id: number;
    description: string;
    transactionDate: string; // ISO string format for DateTime
    value: number;
  
    accountId: number;
  
    categoryId: number;

  
    expenseTypeId: number;

  }

 // src/app/interfaces/expense.ts

export interface ExpenseReceive {
    id: number;
    description: string;
    transactionDate: string; // ISO string format for DateTime
    value: number;
  
    accountId: number;
    account: Account;
  
    categoryId: number;
    category: Category;
  
    expenseTypeId: number;
    expenseType: ExpenseType;
    billImgUrl?: string;
    
  }
  
  export interface Account {
    id: number;
    name: string;
    // Add other properties as per your .NET Account model4
  }
  
  export interface Category {
    id: number;
    name: string;
    // Add other properties as per your .NET Category model
  }
  
  export interface ExpenseType {
    id: number;
    name: string;
    // Add other properties as per your .NET ExpenseType model
  }
  
  
  
  


